vcpkg install <port/features> --only-downloads --keep-going fails with CASCADED_DUE_TO_MISSING_DEPENDENCIES when pre-requisite ports are NOT actually installed

**Describe the bug**
`vcpkg install <port/features> --only-downloads --keep-going` **fails** with CASCADED_DUE_TO_MISSING_DEPENDENCIES when pre-requisite ports are NOT actually installed

**Environment**
- OS: [Microsoft Windows x64 Pro [Version 10.0.18363.778]]
- Compiler:
- Visual Studio 2017 Version: 15.9.22 VS Tools Version: 14.16.27023
- Visual Studio 2019 Version: 16.5.4 VS Tools Version: 14.25.28610
- WinSDK 10.0.18362.0
- cmake version 3.17.1
- ninja version 1.10.0
- Vcpkg package management program version 2020.02.04-nohash-external

**To Reproduce**
Steps to reproduce the behavior:
```
SYSTEM@ANDROMEDA 04/21/2020 21:04:46 D:\Git\vs2017.test
->`vcpkg install sdformat6 urdfdom-headers console-bridge tinyxml urdfdom ignition-modularscripts ignition-cmake0 ignition-math4 --only-downloads --keep-going`
Computing installation plan...
The following packages will be built and installed:
    console-bridge[core]:x64-windows
    ignition-cmake0[core]:x64-windows
    ignition-math4[core]:x64-windows
    ignition-modularscripts[core]:x64-windows
    sdformat6[core]:x64-windows
    tinyxml[core]:x64-windows
    urdfdom[core]:x64-windows
    urdfdom-headers[core]:x64-windows

Starting package 1/8: urdfdom-headers:x64-windows
Building package urdfdom-headers[core]:x64-windows...
-- Using cached D:/Git/vcpkg.downloads/ros-urdfdom_headers-00c1c9c231e46b2300d04073ad696521758fa45c.tar.gz
-- Using source at D:/Git/vs2017.test/buildtrees/urdfdom-headers/src/21758fa45c-13bc02a253
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:43 (message):
  This command cannot be executed in Download Mode.

  Halting portfile execution.

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:297 (vcpkg_execute_required_process)
  ports/urdfdom-headers/portfile.cmake:11 (vcpkg_configure_cmake)
  scripts/ports.cmake:90 (include)

Downloaded sources for package urdfdom-headers[core]:x64-windows
Elapsed time for package urdfdom-headers:x64-windows: 14.61 s

Starting package 2/8: console-bridge:x64-windows
Building package console-bridge[core]:x64-windows...
-- Using cached D:/Git/vcpkg.downloads/ros-console_bridge-3b5b110c224502acdfae621e790caca565173e91.tar.gz
-- Using source at D:/Git/vs2017.test/buildtrees/console-bridge/src/a565173e91-b1e3751f5e
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:43 (message):
  This command cannot be executed in Download Mode.

  Halting portfile execution.

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:297 (vcpkg_execute_required_process)
  ports/console-bridge/portfile.cmake:11 (vcpkg_configure_cmake)
  scripts/ports.cmake:90 (include)

Downloaded sources for package console-bridge[core]:x64-windows
Elapsed time for package console-bridge:x64-windows: 362.1 ms

Starting package 3/8: tinyxml:x64-windows
Building package tinyxml[core]:x64-windows...
-- Note: tinyxml only supports static library linkage. Building static library.
-- Using cached D:/Git/vcpkg.downloads/tinyxml_2_6_2.tar.gz
-- Using source at D:/Git/vs2017.test/buildtrees/tinyxml/src/2.6.2-90b87b703c
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:43 (message):
  This command cannot be executed in Download Mode.

  Halting portfile execution.

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:297 (vcpkg_execute_required_process)
  ports/tinyxml/portfile.cmake:22 (vcpkg_configure_cmake)
  scripts/ports.cmake:90 (include)

Downloaded sources for package tinyxml[core]:x64-windows
Elapsed time for package tinyxml:x64-windows: 389.5 ms

Starting package 4/8: urdfdom:x64-windows
Building package urdfdom[core]:x64-windows...
Error: Building package urdfdom:x64-windows failed with: **CASCADED_DUE_TO_MISSING_DEPENDENCIES**
Elapsed time for package urdfdom:x64-windows: 1.468 ms

Starting package 5/8: ignition-modularscripts:x64-windows
Building package ignition-modularscripts[core]:x64-windows...
Downloaded sources for package ignition-modularscripts[core]:x64-windows
Elapsed time for package ignition-modularscripts:x64-windows: 122 ms

Starting package 6/8: ignition-cmake0:x64-windows
Building package ignition-cmake0[core]:x64-windows...
Error: Building package ignition-cmake0:x64-windows failed with: **CASCADED_DUE_TO_MISSING_DEPENDENCIES**
Elapsed time for package ignition-cmake0:x64-windows: 2.969 ms

Starting package 7/8: ignition-math4:x64-windows
Building package ignition-math4[core]:x64-windows...
Error: Building package ignition-math4:x64-windows failed with: **CASCADED_DUE_TO_MISSING_DEPENDENCIES**
Elapsed time for package ignition-math4:x64-windows: 7.898 ms

Starting package 8/8: sdformat6:x64-windows
Building package sdformat6[core]:x64-windows...
Error: Building package sdformat6:x64-windows failed with: **CASCADED_DUE_TO_MISSING_DEPENDENCIES**
Elapsed time for package sdformat6:x64-windows: 3.673 ms

Total elapsed time: 15.51 s

RESULTS
    urdfdom-headers:x64-windows: DOWNLOADED: 14.61 s
    console-bridge:x64-windows: DOWNLOADED: 362.1 ms
    tinyxml:x64-windows: DOWNLOADED: 389.5 ms
    urdfdom:x64-windows: CASCADED_DUE_TO_MISSING_DEPENDENCIES: 1.468 ms
    ignition-modularscripts:x64-windows: DOWNLOADED: 122 ms
    ignition-cmake0:x64-windows: CASCADED_DUE_TO_MISSING_DEPENDENCIES: 2.969 ms
    ignition-math4:x64-windows: CASCADED_DUE_TO_MISSING_DEPENDENCIES: 7.898 ms
    sdformat6:x64-windows: CASCADED_DUE_TO_MISSING_DEPENDENCIES: 3.673 ms

SUMMARY
    SUCCEEDED: 0
    BUILD_FAILED: 0
    POST_BUILD_CHECKS_FAILED: 0
    FILE_CONFLICTS: 0
    CASCADED_DUE_TO_MISSING_DEPENDENCIES: 4
    EXCLUDED: 0
    DOWNLOADED: 4
```
**Expected behavior**
Before when using the `vcpkg install <ports/features> --only-downloads --keep-going` option vcpkg would download all packages from source site to `<vcpkg-root>\downloads` folder/directory, extract the source to `<vcpkg-root>\buildsystems` folder then performed "Halting portfile execution" and continue on to the next port and download, extract, halt install, repeating for each port listed.

**Failure logs** 
None

**Additional context**
- Has the vcpkg functionality changed where the use --only-downloads cascades failures if pre-requisite ports are not install?
- If true it will be a big step back for those who only wish to download files one an internet connected machine in order to transfer them to a machine without internet or network access?!

Confirmd.
```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:43 (message):
  This command cannot be executed in Download Mode.

  Halting portfile execution.
```
That's expected, but we need to fix error `CASCADED_DUE_TO_MISSING_DEPENDENCIES` in download mode.

--- a/toolsrc/src/vcpkg/build.cpp
+++ b/toolsrc/src/vcpkg/build.cpp
@@ -954,7 +954,7 @@ namespace vcpkg::Build
             }
         }
 
-        if (!missing_fspecs.empty())
+        if (!missing_fspecs.empty() && !Util::Enum::to_bool(action.build_options.only_downloads))
         {
             return {BuildResult::CASCADED_DUE_TO_MISSING_DEPENDENCIES, std::move(missing_fspecs)};
         }

